doc/man/sphinx-apidoc.rst

```diff
@@ -126,29 +126,260 @@ These options are used when :option:`--full` is specified:
 
-   In detail, please refer the system template files Sphinx provides.
-   (``sphinx/templates/apidoc`` and ``sphinx/templates/quickstart``)
+   Please refer the to system template files in ``sphinx/templates/quickstart``
+   for details.
+
+   .. versionadded:: 2.2
+
+
+.. _apidoctemplating:
+
+Templating
+----------
+
+.. versionadded:: 2.2
+   Project templating options for sphinx-apidoc
+
+The TOC template
+~~~~~~~~~~~~~~~~
+
+The template for the TOC file, which is generated unless :option:`--no-toc` is
+given, is in the file ``toc.rst_t``. It uses the following Jinja2 variables:
+
+.. data:: header
+
+   The :confval:`project` name, cf. :option:`-H`
+
+.. data:: maxdepth
+
+   Maximum depth for the generated table of contents file, see :option:`-d`
+
+.. data:: docnames
+
+   A sorted list of the modules in the TOC
+
+The default ``toc.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/toc.rst_t
+   :language: jinja
+
+The package template
+~~~~~~~~~~~~~~~~~~~~
+
+The package template (``package.rst_t``) is used to render packages or implicit
+name spaces. It uses the following Jinja2 variables:
+
+.. data:: pkgname
+
+The fully qualified package name
+
+.. data:: subpackages
+
+List of fully qualified sub-package names (if any)
+
+.. data:: submodules
+
+List of fully qualified sub-module names (if any)
+
+.. data:: is_namespace
+
+Whether or not the template is used to render an implicit name space
+
+.. data:: modulefirst
+
+Whether :option:`--module-first` was given
+
+.. data:: separatemodules
+
+Whether :option:`--separate` was given
+
+.. data:: automodule_options
+
+A comma-separated list of ``automodule`` directives, see
+:envvar:`SPHINX_APIDOC_OPTIONS`.
+
+.. data:: show_headings
+
+True unless :option:`--no-headings` was given
+
+
+The default ``package.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/package.rst_t
+   :language: jinja
+
+
+The module template
+~~~~~~~~~~~~~~~~~~~
+
+The module template is used to render modules. This happens only if
+:option:`--separate` was given, or if *MODULE_PATH* contains standalone
+modules instead of packages.
+
+It uses the :data:`automodule_options` and :data:`show_headings` variables of
+``package.rst_t``, and additionally:
+
+.. data:: qualname
+
+The fully qualified name of the module.
+
+.. data:: basename
+
+An alias for :data:`qualname`.
+
+The default ``module.rst_t`` is
+
+.. literalinclude:: ../../sphinx/templates/apidoc/module.rst_t
+   :language: jinja
+
+
+Accessing module members in templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.3
+   Advanced templating with ``get_members``
+
+Both the ``package.rst_t`` and ``module.rst_t`` templates have access to a
+:func:`get_members` function that allows to extract the members of the package or
+module being rendered:
+
+
+.. py:function:: get_members(fullname, typ=None, include_imported=False, out_format='names', in_list=None, include_private=*private*, known_refs=None)
```

Comment:
@goerz  Is changing `get_members` from a function to a `members` variable that's the simple names of all qualifying members too much a radical simplification?

@tkoyama010 Do you think you would approve and merge such changes?